dotCMS/src/main/webapp/html/portlet/ext/contentlet/field/edit_field.jsp

```diff
@@ -680,24 +680,35 @@
 
             
                 %>
-            <dotcms-binary-field id="binary-field-<%=field.getVelocityVarName()%>"  fieldName="<%=field.getVelocityVarName()%>"></dotcms-binary-field>
+
+            <div id="container-binary-field-<%=field.getVelocityVarName()%>"></div>
             <input name="<%=field.getFieldContentlet()%>" id="binary-field-input-<%=field.getFieldContentlet()%>ValueField" type="hidden" />
 
             <script>
-
                 // Create a new scope so that variables defined here can have the same name without being overwritten.
                 (function autoexecute() {
-                    const binaryField = document.getElementById("binary-field-<%=field.getVelocityVarName()%>");
+                    const binaryFieldContainer = document.getElementById("container-binary-field-<%=field.getVelocityVarName()%>");
                     const field = document.querySelector('#binary-field-input-<%=field.getFieldContentlet()%>ValueField');
+                    const acceptArr = "<%= accept%>".split(',');
+                    const acceptTypes = acceptArr.map((type) => type.trim())
+
+                    // Creating the binary field dynamically
+                    // Help us to set inputs before the ngInit is executed.
+                    const binaryField = document.createElement('dotcms-binary-field');
+                    binaryField.id = "binary-field-<%=field.getVelocityVarName()%>";
+                    binaryField.setAttribute("fieldName", "<%=field.getVelocityVarName()%>")
+                    
                     // Set the initial value.
-                    binaryField.maxFileSize=Number("<%= maxFileLength%>");
-                    binaryField.accept="<%= accept%>";
-                    binaryField.helperText="<%= helperText%>";
-
+                    binaryField.maxFileSize = Number("<%= maxFileLength%>");
```

Comment:
If possible that this can be `NaN`? Should we validate that it is a number?

Reply:
Done!